### PR TITLE
Search portal

### PR DESCRIPTION
**Pull request by alias01**
please review

### CODE REVIEW

**Comment by arrogant-ant** [comment]
@alias01 why images are moved inside backup dir?

File: src/components/molecules/MoleculeSRPCard.vue [hunk 1]
```diff
@@ -0,0 +1,127 @@
+<template>
+  <div class="custom-card">
+    <figure class="card-img image is-96x96">
+      <img :alt="'parking spot image'" :src="spot.IconURL" />
```
Comment (arrogant-ant) [inline src/components/molecules/MoleculeSRPCard.vue:4]: Start alt text with capital letter.
`Parking`

**Review by arrogant-ant — CHANGES_REQUESTED** [review]

File: src/views/PageVOPortal.vue [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        await this.requestSpot();
+        this.isLoading = false;
+        this.$buefy.toast.open({
+          message: "ParkSpot registered successfully!",
+          type: "is-success",
+          duration: 2000,
+        });
+        this.$router.push({ name: "thankYou" });
+      } catch (error) {
+        console.error({ error });
```
Comment (arrogant-ant) [inline src/views/PageVOPortal.vue:40]: if intentional logging then please add some identifier.

**Review by arrogant-ant — CHANGES_REQUESTED** [review]

File: src/components/extras/SearchInput.vue [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  computed: {
+    ...mapGetters({
+      getLocationName: "map/getLocationName",
+    }),
+  },
+  watch: {
+    selected(newValue) {
+      this.updateMapConfig([newValue.center[0], newValue.center[1]]); // needed for recentering of map.
+      this.updateSelectedLocation(newValue.place_name); // get the actual value of selected option.
+      this.updateSelectedCity(newValue.context[0].text); // update selected city
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/extras/SearchInput.vue]: ve

File: src/assets/styles/variables.css [hunk 4]
```diff
@@ -2,4 +2,27 @@
...
+  --border-circle: 1000%;
+  
+  
+  /* typography */
+  --regular-font: 400;
+  --semi-bold-font: 500;
+  --bold-font: 600;
+  --sp-size-sm: 0.75rem;
+  /* - Line height:*/
+    --lh-default:1;
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/assets/styles/variables.css:21]: stylelint introduce in later PR...so not needed now

File: src/components/atoms/AtomSelectInput.vue [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      type: null,
+    },
+  },
+  emits: ["input"],
+  data() {
+    return {
+      innerValue: "",
+    };
+  },
+  // methods: {
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/atoms/AtomSelectInput.vue]: next Pr

File: src/components/atoms/AtomSelectInput.vue [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    placeholder: {
+      type: String,
+      default: "Select anything from list",
+    },
+    /**
+     * The list is list of data
+     * @values Array of Object expecting
+     * e.g. [{id:1, name:'list1'},{id:2, name:'list2'}]
+     */
+    list: {
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/atoms/AtomSelectInput.vue:29]: next PR

File: src/components/atoms/AtomUpload.vue [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        ></button>
+      </span>
+    </div>
+  </section>
+</template>
+
+<script>
+export default {
+  name: "AtomUpload",
+  emits: ["uploadFiles"],
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/atoms/AtomUpload.vue:42]: done

File: src/components/extras/MapContainer.vue [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+      mapConfig: "map/getMapConfig",
+      mapCenter: "map/getNewMapCenter",
+    }),
+  },
+  watch: {
+    mapCenter(newCenter) {
+      this.reCenterMap(newCenter);
+    },
+  },
+  mounted() {
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/extras/MapContainer.vue:42]: next PR

File: src/components/organisms/OrganismPreferenceForm.vue [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+  },
+  watch: {
+    formSubmitted(newVal) {
+      if (newVal) {
+        this.$refs.observer
+          .validate()
+          .then((el) => {
+            if (el) {
+              this.submit();
+              this.$emit("formValidate", el);
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/organisms/OrganismPreferenceForm.vue:93]: done

File: src/components/templates/TemplateNearBy.vue [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+  <div>
+    <HeaderBanner>
+      <AtomHeading class="custom-title"> Book Your Parking Spot </AtomHeading>
+      <b-breadcrumb align="is-centered" size="is-medium">
+        <b-breadcrumb-item tag="router-link" to="/"> Home </b-breadcrumb-item>
+        <b-breadcrumb-item tag="router-link" to="/blog" active>
+          {{ locationWithCaps }}
+        </b-breadcrumb-item>
+      </b-breadcrumb>
+      <!-- <div>
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateNearBy.vue:11]: done

File: src/components/templates/TemplateNearBy.vue [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+          start earning extra.
+        </AtomParagraph>
+      </div>
+    </Wrapper>
+  </div>
+</template>
+
+<script>
+import HeaderBanner from "../extras/HeaderBanner.vue";
+// import SearchInput from "../extras/SearchInput.vue";
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateNearBy.vue:46]: done

File: src/components/templates/TemplateNearBy.vue [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+import AtomHeading from "../atoms/AtomHeading.vue";
+import MoleculeSRPCard from "../molecules/MoleculeSRPCard.vue";
+import AtomParagraph from "../atoms/AtomParagraph.vue";
+import Wrapper from "../extras/Wrapper.vue";
+export default {
+  name: "TemplateNearBy",
+  components: {
+    HeaderBanner,
+    // SearchInput,
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateNearBy.vue:55]: done

File: src/components/templates/TemplateNearBy.vue [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+</script>
+
+<style scoped>
+.custom-title {
+  text-align: center;
+  margin-bottom: 1rem;
+}
+
+/* @media only screen and (max-width: 1024px) {
+  .custom-title {
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateNearBy.vue:103]: done

File: src/components/templates/TemplateOurProducts.vue [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+        },
+      },
+    };
+  },
+  methods: {
+    info(value) {
+      this.test = value;
+    },
+    onArrowBtn() {
+      this.$emit("arrowBtn");
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateOurProducts.vue:77]: task created for this in rock

File: src/components/templates/TemplateSOPortal.vue [hunk 15]
```diff
@@ -0,0 +1,218 @@
...
+  name: "TemplateSOPortal",
+  components: {
+    OrganismContactForm,
+    OrganismKycForm,
+    OrganismAdditionalInfo,
+    OrganismAddressForm,
+    AtomHeading,
+    Wrapper,
+  },
+  emits: ["finalSubmit"],
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateSOPortal.vue:123]: task created

File: src/components/templates/TemplateSrp.vue [hunk 16]
```diff
@@ -0,0 +1,110 @@
+<template>
+  <div class="srp-container">
+    <div class="srp-lists">
+      <Pagination :totals="totals" @changed="onPageChange"></Pagination>
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateSrp.vue:4]: done

File: src/components/templates/TemplateThankYou.vue [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+import AtomButton from "../atoms/AtomButton.vue";
+export default {
+  name: "TemplateThankYou",
+  components: {
+    Wrapper,
+    AtomImage,
+    AtomParagraph,
+    AtomButton,
+  },
+  emits: ["homeBtn"],
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateThankYou.vue:36]: task created

File: src/components/templates/TemplateVOPortal.vue [hunk 18]
```diff
@@ -0,0 +1,202 @@
...
+        :icon-prev="prevIcon"
+        :icon-next="nextIcon"
+        :label-position="labelPosition"
+        :mobile-mode="mobileMode"
+      >
+        <b-step-item
+          step="1"
+          label="Step 1"
+          :clickable="isStepsClickable"
+          :type="btnStack[0] ? 'is-success' : 'is-warning'"
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateVOPortal.vue:19]: will think about it

File: src/components/templates/TemplateVOPortal.vue [hunk 19]
```diff
@@ -0,0 +1,202 @@
...
+export default {
+  name: "TemplateVOPortal",
+  components: {
+    OrganismContactForm,
+    OrganismPreferenceForm,
+    OrganismAddressForm,
+    AtomHeading,
+    Wrapper,
+  },
+  emits: ["finalSubmit"],
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateVOPortal.vue:106]: task created for this already

File: src/inlcudes/ConvertImageToByte.js [hunk 20]
```diff
@@ -0,0 +1,24 @@
+const reader = new FileReader();
+const fileByteArray = [];
+
+function getConvertedImage(data) {
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/inlcudes/ConvertImageToByte.js:4]: done

File: src/main.js [hunk 21]
```diff
@@ -1,38 +1,39 @@
...
-import "bulma/css/bulma.css";
 import store from "./store";
-import ClickOutside from "vue-click-outside";
 import fbApp from "./FB-App.vue";
 
-Vue.component("jw-pagination", JwPagination);
-Vue.directive("lazyload", LazyLoadDirective);
-Vue.directive("ClickOutside", ClickOutside);
-Vue.use(VueResource);
+// import VueResource from "vue-resource";
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/main.js:7]: done

File: src/main.js [hunk 22]
```diff
@@ -1,38 +1,39 @@
...
-  // A CSS class applied to elements with the
-  // v-scroll-reveal directive; useful for animation overrides.
-  class: "v-scroll-reveal",
-  duration: 1000,
-  scale: 1,
-  distance: "20px",
-  mobile: true,
-});
+Vue.use(Buefy);
+// Vue.use(VueResource);
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/main.js:17]: done

File: src/services/api.js [hunk 23]
```diff
@@ -110,6 +120,18 @@ class MayaApiService extends BaseApiService {
   }
 }
 
+class MapBoxApiService extends BaseApiService {
+  constructor() {
+    let mapBoxDomain = "https://api.mapbox.com"; //TODO: we can pick from .env files.
+    let baseHeaderMap = {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      // Flavour: flavour,
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/services/api.js:129]: done

**Review by arrogant-ant — APPROVED** [review]

**Pull request closed without merging**